src/linux/classes/com/alibaba/wisp/engine/WispEngine.java

```diff
@@ -0,0 +1,590 @@
+                    @Override
+                    public Thread newThread(Runnable r) {
+                        Thread thread = new Thread(WispEngine.daemonThreadGroup, r);
+                        thread.setDaemon(true);
+                        thread.setName("Wisp-Timer");
+                        return thread;
+                    }
+                });
+    }
+
+    private static void initializeWispClass() {
+        assert JLA != null : "WispCarrier should be initialized after System";
+        assert JLA.currentThread0().getName().equals("main") : "Wisp need to be loaded by main thread";
+        shiftThreadModel = WispConfiguration.ENABLE_THREAD_AS_WISP;
+        daemonThreadGroup = new ThreadGroup(JLA.currentThread0().getThreadGroup(), "Daemon Thread Group");
+        carrierThreads = new ConcurrentSkipListSet<>(new Comparator<Thread>() {
+            @Override
+            public int compare(Thread o1, Thread o2) {
+                return Long.compare(o1.getId(), o2.getId());
+            }
+        });
+        WISP_ROOT_ENGINE = new WispEngine(WISP_ROOT_ENGINE_NAME);
+        if (transparentWispSwitch()) {
+            WispEngine.initializeClasses();
+            JLA.wispBooted();
+        }
+    }
+
+    private static void initializeClasses() {
+        try {
+            Class.forName(CoroutineExitException.class.getName());
+            Class.forName(WispThreadWrapper.class.getName());
+            Class.forName(TaskDispatcher.class.getName());
+            Class.forName(StartShutdown.class.getName());
+            Class.forName(NotifyAndWaitTasksForShutdown.class.getName());
+            Class.forName(Coroutine.StealResult.class.getName());
+            Class.forName(WispCounterMXBeanImpl.class.getName());
+            Class.forName(ThreadAsWisp.class.getName());
+            Class.forName(WispEventPump.class.getName());
+            if (WispConfiguration.WISP_PROFILE) {
+                Class.forName(WispPerfCounterMonitor.class.getName());
+            }
+            if (WispConfiguration.WISP_HIGH_PRECISION_TIMER) {
+                timer.submit(new Runnable() {
+                    @Override
+                    public void run() {
+                    }
+                });
+            }
+            new ConcurrentLinkedQueue<>().iterator();
+            new ConcurrentSkipListMap<>().keySet().iterator();
+            WispCarrier carrier = WispCarrier.current();
+            carrier.addTimer(System.nanoTime() + Integer.MAX_VALUE, false);
+            carrier.cancelTimer();
+            carrier.createResumeEntry(new WispTask(carrier, null, false, false));
+            registerPerfCounter(carrier);
+            deRegisterPerfCounter(carrier);
+        } catch (Exception e) {
+            throw new ExceptionInInitializerError(e);
+        }
+    }
+
+    private static void startWispDaemons() {
+        if (transparentWispSwitch()) {
+            pollerThread = new Thread(daemonThreadGroup, new Runnable() {
```

Comment:
pollerThread --> unparkDispatcher

please include the polish patch

Reply:
Done.